ouroboros-consensus/src-win32/Ouroboros/Storage/IO.hs

```diff
@@ -74,6 +75,18 @@ read fh bytes = withOpenHandle "read" fh $ \h ->
   Internal.createUptoN (fromIntegral bytes) $ \ptr ->
     fromIntegral <$> win32_ReadFile h ptr (fromIntegral bytes) Nothing
 
+getCurrentFileOffset :: HANDLE -> IO Int64
+getCurrentFileOffset h = setFilePointerEx h 0 fILE_CURRENT
+
+pread :: FHandle -> Word64 -> Int64 -> IO ByteString
+pread fh bytes pos = withOpenHandle "pread" fh $ \h ->
+  Internal.createUptoN (fromIntegral bytes) $ \ptr -> do
+    initialOffset <- getCurrentFileOffset h
+    _ <- setFilePointerEx h pos fILE_BEGIN
```

Comment:
What if it fails? Does it throw an error or does it report a special return value?

Reply:
The Haskell wrappers take any failing error code and throw it with `throwIO`.